### PR TITLE
release 0.12: add changelog

### PR DESCRIPTION
**Pull request by smelc**
Release checklist: https://github.com/tweag/rules_haskell/issues/1278

For the record, I gathered the list of PRs since release 0.11 here: https://github.com/tweag/rules_haskell/pulls?q=is%3Apr+merged%3A%3E%3D2019-10-10

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,14 +6,71 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/).
...
+## Removed
+
+* `hazel` has been deleted, please use `stack_snapshot` instead.
+
+## Changed
 
 * The `haskell_register_toolchains()` is no longer defined in
   `haskell/repositories.bzl`, load it from `haskell/toolchain.bzl` instead.
+* `cabal` wrapper: specify `python3` as a requirement,
+  to enhance error messages on macOS
```
Comment (aherrmann) [inline CHANGELOG.md]: Also to reduce cache invalidations, see https://github.com/tweag/rules_haskell/pull/1097 and https://github.com/tweag/rules_haskell/issues/1096

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,14 +6,71 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/).
...
+  See [#1208](https://github.com/tweag/rules_haskell/pull/1208).
+* `haskell_cabal_library` and `stack_snapshot` now have a
+  Boolean parameter `haddock` to specify whether haddock generation
+  should be performed (defaulted to `True`).
+* Added support for GHC bindist versions `8.8.3`, `8.8.2`, and `8.8.1`.
+* Windows: support for `cabal`
+  See [#1133](https://github.com/tweag/rules_haskell/pull/1133).
+* `haskell_cabal_library` now builds and exports haddock documentation
+  See [#1102](https://github.com/tweag/rules_haskell/pull/1102).
+
```
Comment (aherrmann) [inline CHANGELOG.md]: The additional dependency on `rules_sh` seems worth mentioning, see https://github.com/tweag/rules_haskell/pull/1136 and https://github.com/tweag/rules_haskell/pull/1117 and https://github.com/tweag/rules_haskell/pull/1143 for motivation

File: CHANGELOG.md [hunk 3]
```diff
@@ -6,14 +6,71 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/).
...
+## Removed
+
+* `hazel` has been deleted, please use `stack_snapshot` instead.
```
Comment (aherrmann) [inline CHANGELOG.md]: AFAIK Hazel was never officially supported, but I see it was mentioned in 0.9 so it seems reasonable to list it here.

File: CHANGELOG.md [hunk 4]
```diff
@@ -6,14 +6,71 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/).
...
+### [0.12.0] - 2020-03-DD TODO set day
 
-### Changed
+## Highlights
+
+* Various improvements to Windows support.
+
+* Support for Bazel 2.0.0
+
+* Minimum supported Bazel version is now 0.29
```
Comment (aherrmann) [inline CHANGELOG.md]: At least on MacOS with GHC bindist Bazel 0.29 is failing, see [this CircleCI run](https://circleci.com/gh/tweag/rules_haskell/8684?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-checks-link).

File: CHANGELOG.md [hunk 5]
```diff
@@ -6,14 +6,71 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/).
...
+* `stack_snapshot`: warning of stackage dependencies are not shown anymore,
+  as they are irrelevant.
+  See [#1146](https://github.com/tweag/rules_haskell/pull/1146) and
+  [#1026](https://github.com/tweag/rules_haskell/issues/1026).
+* Sorted the content of generated manifest files, hereby avoid some spurious
+  rebuilds (more builds caching).
+  See [#1128](https://github.com/tweag/rules_haskell/pull/1128) and
+  [#1126](https://github.com/tweag/rules_haskell/issues/1126).
+
+## Fixed
```
Comment (aherrmann) [inline CHANGELOG.md]: Also fixed some issues with `stack update` racing on a lock file which may be worth mentioning, see https://github.com/tweag/rules_haskell/pull/1199

File: CHANGELOG.md [hunk 6]
```diff
@@ -6,14 +6,71 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/).
...
+* Various improvements to Windows support.
+
+* Support for Bazel 2.0.0
+
+* Minimum supported Bazel version is now 0.29
+
+## Added
+
+* `haskell_repl` now has a `hie_bios` output group
+* Added support for [hrepl](https://github.com/google/hrepl)
```
Comment (aherrmann) [inline CHANGELOG.md]: The PR has some nice details, so would be good to link to it here https://github.com/tweag/rules_haskell/pull/1210

File: CHANGELOG.md [hunk 7]
```diff
@@ -6,14 +6,71 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/).
...
+* `cabal` wrapper: specify `python3` as a requirement,
+  to enhance error messages on macOS
+* `ghc_bindist` and `haskell_register_ghc_bindists` now have
+  a `locale` argument. Set it to circumvent issues on systems without
+  the default `C.UTF-8` locale.
+  See [#1249](https://github.com/tweag/rules_haskell/pull/1249).
+* macOS: `BAZEL_USE_CPP_ONLY_TOOLCHAIN = 1` must be set for Bazel
+  to pick the correct C compiler.
+  See [#1159](https://github.com/tweag/rules_haskell/pull/1159).
+  See [#1158](https://github.com/tweag/rules_haskell/pull/1158).
```
Comment (aherrmann) [inline CHANGELOG.md]: #1158 is the delete Hazel PR.

File: CHANGELOG.md (hunk 1 above)
Comment (smelc) [reply CHANGELOG.md]: Added in cb537c2.

File: CHANGELOG.md (hunk 2 above)
Comment (smelc) [reply CHANGELOG.md]: Added an item at the end of the _Added_ section in cb537c2.

File: CHANGELOG.md (hunk 4 above)
Comment (smelc) [reply CHANGELOG.md]: I've modified the _Highlights_ section accordingly in cb537c2.

File: CHANGELOG.md (hunk 5 above)
Comment (smelc) [reply CHANGELOG.md]: I added an item in the _Changed_ section in cb537c2. It could have been in _Fixed_ too.

File: CHANGELOG.md (hunk 6 above)
Comment (smelc) [reply CHANGELOG.md]: Done in cb537c2.

File: CHANGELOG.md (hunk 7 above)
Comment (smelc) [reply CHANGELOG.md]: I've mentioned it in cb537c2.

**Comment by smelc** [comment]
> A couple comments, and a few entries are missing links to the corresponding PRs.

Thanks @aherrmann, I double checked that all entries have links to PRs starting at cb537c2. I only left them out for the entry mentioning the addition of the three GHC bindist versions. It seemed overkill.

File: CHANGELOG.md [hunk 8]
```diff
@@ -6,14 +6,92 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/).
...
+
+* `haskell_repl` now has a `hie_bios` output group
+  See [#1263](https://github.com/tweag/rules_haskell/pull/1263)
+* Added support for [hrepl](https://github.com/google/hrepl)
+  (a standalone binary that runs REPLS for Bazel Haskell targets).
+  See [#1210](https://github.com/tweag/rules_haskell/pull/1210).
+* `haskell_cabal_library`, `haskell_cabal_binary`, and `stack_snapshot`
+  now have a `verbose` argument, to allow suppressing their output.
+  See [#1208](https://github.com/tweag/rules_haskell/pull/1208).
+* `haskell_cabal_library` and `stack_snapshot` now have a
```
Comment (regnat) [inline CHANGELOG.md]: I think this and https://github.com/tweag/rules_haskell/pull/1279/files#diff-4ac32a78649ca5bdd8e0ba38b7006a1eR40 could be merged into one entry as #1200 is just about making #1102 optional. Maybe something like

```markdown
 * `haskell_cabal_library` and `stack_snapshot` now build the haddock documentation if the parameter `haddock` is set to `True` (the default).
  See [#1102](https://github.com/tweag/rules_haskell/pull/1102) and  [#1200](https://github.com/tweag/rules_haskell/pull/1200)

Comment (smelc) [reply CHANGELOG.md]: Implemented this suggestion in 1191ebf

File: CHANGELOG.md (hunk 7 above)
Comment (aherrmann) [reply CHANGELOG.md]: Sorry, I wasn't very clear. What I meant is, you're listing #1158 under the `BAZEL_USE_CPP_ONL_TOOLCHAIN` item. This is unrelated to Hazel, so shouldn't be listed here. (Unless I'm missing something).

Comment (smelc) [reply CHANGELOG.md]: Oh I get your point now :-) I removed the mention to #1158 in the `BAZEL_USE_CPP_ONL_TOOLCHAIN` item in https://github.com/tweag/rules_haskell/commit/ea3f13165dae68d5c0291ef258fb1cd51ea0372d#diff-4ac32a78649ca5bdd8e0ba38b7006a1eR66-R69 in commit ea3f13165dae68d5c0291ef258fb1cd51ea0372d.

**Pull request merged**